File: bench/07_string_build/bench.py
import time

n = 5000
t0 = time.time()
parts = []
for i in range(n):
    parts.append(f"item-{i};")
s = "".join(parts)
ms = (time.time() - t0) * 1000
print(f"result: {len(s)}  time: {ms:.0f}ms")

File: bench/07_string_build/bench.rs
use std::time::Instant;

fn main() {
    let n = 5000;
    let t0 = Instant::now();
    let mut s = String::new();
    for i in 0..n {
        s.push_str(&format!("item-{};", i));
    }
    let ms = t0.elapsed().as_millis();
    println!("result: {}  time: {}ms", s.len(), ms);
}
